### PR TITLE
Feat/Add tree cmds to CLI

### PR DESCRIPTION
**Pull request by carpawell**
Related to #1332.

### CODE REVIEW

File: pkg/services/tree/signature.go [hunk 1]
```diff
@@ -143,7 +143,7 @@ func verifyMessage(m message) error {
 	return nil
 }
 
-func signMessage(m message, key *ecdsa.PrivateKey) error {
+func SignMessage(m message, key *ecdsa.PrivateKey) error {
```
Comment (fyrchik) [inline pkg/services/tree/signature.go:146]: It needs a comment now.

File: cmd/neofs-cli/modules/tree/add.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+	pk := key.GetOrGenerate(cmd)
+
+	var cnr cid.ID
+	err := cnr.DecodeString(cmd.Flag(cidFlagKey).Value.String())
+	common.ExitOnErr(cmd, "decode container ID string: %w", err)
+
+	tid := cmd.Flag(tidFlagKey).Value.String()
+	if tid == "" {
+		common.ExitOnErr(cmd, "", errors.New("empty tree ID"))
+	}
```
Comment (fyrchik) [inline cmd/neofs-cli/modules/tree/add.go:48]: Can we just make the flag required? I believe tree service supports empty names now.

File: cmd/neofs-cli/modules/tree/add.go [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+	rawCID := make([]byte, sha256.Size)
+	cnr.Encode(rawCID)
+
+	req := new(tree.AddRequest)
+	req.Body = &tree.AddRequest_Body{
+		ContainerId: rawCID,
+		TreeId:      tid,
+		ParentId:    pid,
+		Meta:        meta,
+		BearerToken: nil, // TODO: add token handling
```
Comment (fyrchik) [inline cmd/neofs-cli/modules/tree/add.go:69]: Issue?

File: cmd/neofs-cli/modules/tree/add.go [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+		pairs = append(pairs, &pair)
+	}
+
+	return pairs, nil
+}
+
+func bytesFromString(s string) ([]byte, error) {
+	switch {
+	case strings.HasPrefix(s, "0x"):
```
Comment (fyrchik) [inline cmd/neofs-cli/modules/tree/add.go:110]: Do we need this in CLI? Especially when we don't have similar logic in get operations now.

File: cmd/neofs-cli/modules/tree/root.go [hunk 5]
```diff
@@ -21,6 +23,11 @@ const (
 	pidFlagKey = "pid"
 
 	metaFlagKey = "meta"
+
+	pathFlagKey          = "path"
+	pathAttributeFlagKey = "pattr"
+
+	latestOnlyFlagKey = "latest-only"
```
Comment (fyrchik) [inline cmd/neofs-cli/modules/tree/root.go:32]: To me `--latest` looks more convenient.

File: cmd/neofs-cli/modules/tree/get_by_path.go [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+
+	common.ExitOnErr(cmd, "message signing: %w", tree.SignMessage(req, pk))
+
+	resp, err := cli.GetNodeByPath(ctx, req)
+	common.ExitOnErr(cmd, "rpc call: %w", err)
+
+	nn := resp.GetBody().GetNodes()
+	if len(nn) == 0 {
+		common.ExitOnErr(cmd, "", errors.New("node not found"))
+	}
```
Comment (fyrchik) [inline cmd/neofs-cli/modules/tree/get_by_path.go:89]: I don't think it is an error, it is a valid response.

Comment (carpawell) [reply cmd/neofs-cli/modules/tree/get_by_path.go:89]: do you mean exit with `0` not with the `1`?

File: cmd/neofs-cli/modules/tree/add.go (hunk 4 above)
Comment (carpawell) [reply cmd/neofs-cli/modules/tree/add.go:110]: > Do we need this in CLI

i do not have info about how values in the Tree's KV gonna (planned to) be used, so could not answer that question at all

> Especially when we don't have similar logic in get operations now.

i am ready to add that but dont know how that is expected to work (how to distinguish string bytes from the raw bytes)

in a word -- dropped

File: cmd/neofs-cli/modules/tree/add.go (hunk 3 above)
Comment (carpawell) [reply cmd/neofs-cli/modules/tree/add.go:69]: added #1891

File: cmd/neofs-cli/modules/tree/add.go (hunk 2 above)
Comment (carpawell) [reply cmd/neofs-cli/modules/tree/add.go:48]: > I believe tree service supports empty names now

ok, but what for? dropped that limitation

File: pkg/services/tree/signature.go (hunk 1 above)
Comment (carpawell) [reply pkg/services/tree/signature.go:146]: thought about that but racked my brains what words should be here

added, check, please

File: cmd/neofs-cli/modules/tree/get_by_path.go (hunk 6 above)
Comment (fyrchik) [reply cmd/neofs-cli/modules/tree/get_by_path.go]: Yes, an we can print nothing in this case. Maybe print `node not found` in verbose mode?

File: cmd/neofs-cli/modules/tree/root.go [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+	initAddCmd()
+	initGetByPathCmd()
+	initAddByPathCmd()
+}
+
+const (
+	cidFlagKey = "cid"
+	tidFlagKey = "tid"
+	pidFlagKey = "pid"
```
Comment (fyrchik) [inline cmd/neofs-cli/modules/tree/root.go:25]: Not immediately obvious, maybe `parentIDFlagKey`?

**Review by fyrchik — APPROVED** [review]

File: cmd/neofs-cli/modules/tree/add_by_path.go [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+	"github.com/nspcc-dev/neofs-node/cmd/neofs-cli/internal/key"
+	"github.com/nspcc-dev/neofs-node/pkg/services/tree"
+	cid "github.com/nspcc-dev/neofs-sdk-go/container/id"
+	"github.com/nspcc-dev/neofs-sdk-go/object"
+	"github.com/spf13/cobra"
+)
+
+var addByPathCmd = &cobra.Command{
+	Use:   "add-by-path",
+	Short: "Get a node by its path",
```
Comment (acid-ant) [inline cmd/neofs-cli/modules/tree/add_by_path.go:18]: Get --> Add

**Review by acid-ant — APPROVED** [review]

**Pull request merged**